core/string/ustring.h

```diff
@@ -385,9 +385,9 @@ class String {
 	char32_t unicode_at(int p_idx) const;
 
 	CharString ascii(bool p_allow_extended = false) const;
-	CharString utf8() const;
-	Error parse_utf8(const char *p_utf8, int p_len = -1, bool p_skip_cr = false);
-	static String utf8(const char *p_utf8, int p_len = -1);
+	CharString utf8(bool p_esc_null = true) const;
```

Comment:
Instead of `p_esc_null`, it might be better to add a new set of `cesu8` functions (which will also encode/decode surrogates instead of real character codes), otherwise it's some sort of non-standard encoding, in between of UTF-8 and CESU-8.

Reply:
Part of me wants to avoid a wholesale implementation, as this encoding would only be used internally. With that said, I haven't looked too into the specifics of `CESU-8`, so it's entirely possible that it'd be a reasonable addition

Reply:
Actually… It's entirely possible that this escape format *won't* be needed. I'll have to do more tests, but it seems like a lot of the systems work as expected thanks to the other changes